packages/wrangler/src/type-generation/index.ts

```diff
@@ -2,150 +2,141 @@ import * as fs from "node:fs";
+import { findWranglerToml } from "../config";
+import { defineCommand } from "../core";
 import { getEntry } from "../deployment-bundle/entry";
 import { getVarsForDev } from "../dev/dev-vars";
 import { CommandLineArgsError, UserError } from "../errors";
 import { logger } from "../logger";
 import { parseJSONC } from "../parse";
-import { printWranglerBanner } from "../update-check";
 import { generateRuntimeTypes } from "./runtime";
 import { logRuntimeTypesMessage } from "./runtime/log-runtime-types-message";
 import type { Config } from "../config";
 import type { CfScriptFormat } from "../deployment-bundle/worker";
-import type {
-	CommonYargsArgv,
-	StrictYargsOptionsToInterface,
-} from "../yargs-types";
-
-export function typesOptions(yargs: CommonYargsArgv) {
-	return yargs
-		.positional("path", {
-			describe: "The path to the declaration file to generate",
+
+defineCommand({
+	command: "wrangler types",
+	metadata: {
+		description:
+			"📝 Generate types from bindings and module rules in configuration\n",
+		owner: "Workers: Authoring and Testing",
+		status: "stable",
+	},
+	args: {
+		path: {
 			type: "string",
+			describe: "The path to the declaration file to generate",
 			default: "worker-configuration.d.ts",
 			demandOption: false,
-		})
-		.option("env-interface", {
+		},
+		"env-interface": {
 			type: "string",
 			default: "Env",
 			describe: "The name of the generated environment interface",
 			requiresArg: true,
-		})
-		.option("experimental-include-runtime", {
+		},
+		"experimental-include-runtime": {
 			alias: "x-include-runtime",
 			type: "string",
 			describe: "The path of the generated runtime types file",
 			demandOption: false,
-		});
-}
-
-export async function typesHandler(
-	args: StrictYargsOptionsToInterface<typeof typesOptions>
-) {
-	const { envInterface, path: outputPath } = args;
+		},
+	},
+	positionalArgs: ["path"],
+	async handler(args, { config }) {
+		const { envInterface, path: outputPath } = args;
 
-	const validInterfaceRegex = /^[a-zA-Z][a-zA-Z0-9_]*$/;
+		const validInterfaceRegex = /^[a-zA-Z][a-zA-Z0-9_]*$/;
 
-	if (!validInterfaceRegex.test(envInterface)) {
-		throw new CommandLineArgsError(
-			`The provided env-interface value ("${envInterface}") does not satisfy the validation regex: ${validInterfaceRegex}`
-		);
-	}
+		if (!validInterfaceRegex.test(envInterface)) {
+			throw new CommandLineArgsError(
+				`The provided env-interface value ("${envInterface}") does not satisfy the validation regex: ${validInterfaceRegex}`
+			);
+		}
 
-	if (!outputPath.endsWith(".d.ts")) {
-		throw new CommandLineArgsError(
-			`The provided path value ("${outputPath}") does not point to a declaration file (please use the 'd.ts' extension)`
-		);
-	}
+		if (!outputPath.endsWith(".d.ts")) {
+			throw new CommandLineArgsError(
+				`The provided path value ("${outputPath}") does not point to a declaration file (please use the 'd.ts' extension)`
+			);
+		}
```

Comment:
The difference you see on the tests seems to be caused by us calling `readConfig` before the handler. Move these logic to `validateArgs` might fix it

Reply:
ugh it isn't failing locally, that's super weird - but good point